modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -166,6 +180,31 @@ object CurrencySnapshotAcceptanceManager {
           .getOrElse(acc)
       }
 
+    private def acceptFeeTxs(
+      balances: SortedMap[Address, Balance],
+      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
+    ): F[(SortedMap[Address, Balance], Option[SortedSet[Signed[FeeTransaction]]])] =
+      maybeTxs match {
+        case Some(txs) =>
+          txs.foldLeftM((balances, SortedSet.empty[Signed[FeeTransaction]].some)) { (acc, tx) =>
+            val (balances, txs) = acc
+
+            (
+              balances.getOrElse(tx.source, Balance.empty).minus(tx.amount),
+              balances.getOrElse(tx.destination, Balance.empty).plus(tx.amount)
+            ).mapN {
+              case (src, dst) =>
+                val updatedBalances = balances
+                  .updated(tx.source, src)
+                  .updated(tx.destination, dst)
+                val acceptedTxs = txs.map(_ + tx)
+                (updatedBalances, acceptedTxs)
+            }.fold(e => new Throwable(s"Unexpected state when applying fee transactions: ${e.show}").raiseError, _.pure[F])
+          }
+        case None =>
+          (balances, maybeTxs).pure[F]
+      }
```

Comment:
1. I think this will abort on the first fee transaction that fails. Is that ok?
2. What do you think about this refactor?

```suggestion
    private def acceptFeeTxs(
      balances: SortedMap[Address, Balance],
      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
    ): F[(SortedMap[Address, Balance], Option[SortedSet[Signed[FeeTransaction]]])] =
      maybeTxs match {
        case None => (balances, maybeTxs).pure[F]
        case Some(txs) =>
          txs
            .foldLeftM((balances, SortedSet.empty[Signed[FeeTransaction]])) {
              case ((balances, txs), tx) =>
                (
                  balances.getOrElse(tx.source, Balance.empty).minus(tx.amount).liftTo[F],
                  balances.getOrElse(tx.destination, Balance.empty).plus(tx.amount).liftTo[F]
                ).mapN {
                  case (src, dst) =>
                    val updatedBalances = balances
                      .updated(tx.source, src)
                      .updated(tx.destination, dst)
                    (updatedBalances, txs + tx)
                }
                  .adaptErr(e => new ArithmeticException(s"Unexpected state when applying fee transaction ${tx.show}: ${e.getMessage}"))
            }
            .map { case (m, s) => (m, s.some) }
      }
```

Reply:
>I think this will abort on the first fee transaction that fails. Is that ok?

If fee transactions are being accepted in order then it is the correct way to handle it since it is an unexpected state caused by bad implementation of data application. Alternatively we could just keep applying balances (treated as just non-refined `Long`) and just verify at the very end if any of these `Long`s is negative (doesn't fit `NonNegLong`). Maybe the alternative way is better 🤔

Reply:
>What do you think about this refactor?
I like the `ArithmeticException` one :+1:

Reply:
I have implemented the alternative version mentioned above^